source/sigv4.c

```diff
@@ -2921,6 +2964,38 @@ static SigV4Status_t generateSigningKey( const SigV4Parameters_t * pSigV4Params,
     return returnStatus;
 }
 
+static SigV4Status_t writePayloadHashToCanonicalRequest( const SigV4Parameters_t * pParams,
+                                                         CanonicalContext_t * pCanonicalContext )
+{
+    size_t encodedLen = 0U;
+    SigV4Status_t returnStatus = SigV4Success;
+
+    assert( pParams != NULL );
+    assert( pCanonicalContext != NULL );
+
+    if( FLAG_IS_SET( pParams->pHttpParameters->flags, SIGV4_HTTP_PAYLOAD_IS_HASH ) )
+    {
+        /* Copy the hashed payload data supplied by the user in the headers data list. */
+        returnStatus = copyHeaderStringToCanonicalBuffer( pCanonicalContext->pHashPayloadLoc, pCanonicalContext->hashPayloadLen, pParams->pHttpParameters->flags, '\n', pCanonicalContext );
+        pCanonicalContext->pBufCur--;
```

Comment:
What's the decrement for?

Reply:
We don't need new line but the above function adds new line also.